zombies.js

```diff
@@ -79,17 +97,115 @@
  * @property {method} getMaxHealth         Returns private variable `maxHealth`.
  */
 
+class Player {
+  constructor(name, health, strength, speed){
+    this._pack = [];
+    this._maxHealth = health;
+    this.name = name;
+    this.health = health;
+    this.strength = strength;
+    this.speed = speed;
+    this.isAlive = true;
+    this.equipped = false;
+  }
+  getPack(){
+    return this._pack;
+  }
+  getMaxHealth(){
+    return this._maxHealth = 100;
+  }
+  checkPack(){
+    console.log(this.getPack());
+    return this.getPack();
+  }
+  takeItem(item, weapon, food){
+    if(this._pack.length < 3){
+      this.name = name;
+      this._pack.push(item);
+      console.log('Nice! ' + this.name + 'added ' + this.item + ' to the pack.');
+      return true;
+    }else{
+      console.log('The pack is full so the item could not be stored');
+      return false;
+    }
+  }
+  discardItem(item){
+    if(this._pack.indexOf(item) === -1){
+      console.log(this.item + ' could not be found. No ' + this.item + 'discarded.');
+      return false;
+    }else{
+      this.getPack().splice(this._pack.indexOf(item), 1);
+      console.log(this.name + ', ' + this.item + ' was discarded.');
+      return true;
+    }
+  }
+  equip(itemToEquip){
+    var weaponToUse = this._pack.indexOf(itemToEquip);
+
+    if(itemToEquip instanceof Weapon  &&  weaponToUse > -1 && !this.equipped){
+        this.equipped = itemToEquip;
+        this._pack.splice(weaponToUse, 1);
+
+      }else if( this.equipped !== false ){
+        let newWeapon =this.equipped;
+        this.equipped = itemToEquip;
+        this._pack.splice(weaponToUse, 1, newWeapon);
+      }
+    }
+
+   eat(itemToEat){
+     let findFood = this._pack.indexOf(itemToEat)
+
+     if(itemToEat instanceof Food && findFood > -1 && (this.health + itemToEat.energy) > this.getMaxHealth()){
```

Comment:
Dont put too much in the conditional test.
You can check if it is found and a Food first. then inside the body you can worry about if maxHeath is exceeded.